### ISSUE TITLE
configurable timeout

### ISSUE THREAD

**Issue by JarbasAl** [OpenVoiceOS/ovos-tts-server-plugin#15]
I thought about that, opted for 30 in case someone is using TTS with a very high RTF, even though 10 is probably already too long. It should probably be configurable - maybe an issue for a next PR?

_Originally posted by @mikejgray in https://github.com/OpenVoiceOS/ovos-tts-server-plugin/pull/14#discussion_r1439091615_

**Comment by emphasize** [comment]
should be acting in time, as the plugin also is used on personal servers which are supposed to synth 30 seconds+

**Comment by mikejgray** [comment]
@emphasize We have the timeout configurable now, but to your point, it should be smarter. What would you suggest for an algorithm?

Assuming 6-7 wps (I timed it on default apope voice from Piper):

`((len(text.split(" ") / 6.5) * average_rtf) * 1.2`

That would give us a ballpark expectation of how long it should take, then give yourself 20% on top, although it doesn't account for latency or busy servers or anything else outside of that static 20% buffer.

Or do we even need to get this granular? I'm also not sure if average RTF is a metric available to this plugin from the servers or TTS plugins.

@JarbasAl would probably know off the top of his head

**Comment by builderjer** [comment]
How would you know how many words are being returned to calculate the time out?  You could ask for the temp and it could return "56.5 degrees" or "Currently the temperature in Hotchkiss is 55.6 degrees"

**Comment by mikejgray** [comment]
> How would you know how many words are being returned to calculate the time out? You could ask for the temp and it could return "56.5 degrees" or "Currently the temperature in Hotchkiss is 55.6 degrees"

This is why I'm asking - there are a lot of unknowns here, and we'd need to make sure we're gathering the appropriate telemetry to calculate this if that's how we want to do it. We'll know how many words are being returned because the exact text string goes to TTS, right?

**Comment by builderjer** [comment]
yes, now that I think of it.  It knows the string before it gets sent.  That makes sense.  

have you tried it with your algorithm above?

**Comment by mikejgray** [comment]
> yes, now that I think of it. It knows the string before it gets sent. That makes sense.
> 
> have you tried it with your algorithm above?

Nope! I wanted to get @emphasize and @JarbasAl 's feedback before I start pushing code out anywhere. I'm not exactly sure how much value dynamic adjustment of the timeout gives us, and the risk is a poor user experience. I'm also not sure this is even what Swen was talking about so I wanted his clarification before coding anything.

**Comment by emphasize** [comment]
sry. Just back from a short skiing trip, but appearently caught in time.

> ((len(text.split(" ") / 6.5) * average_rtf) * 1.2

something along those lines, but instead of an average rtf a config value (`average_rtf` being a nice candidate for the default)

_/ edit: but it all gets murky if we request multiple server (per server?)_

**Comment by emphasize** [comment]
STT timeout would be more accurate:
the spec of the mic (which _should_(?) be a part of the passed tts config), so
`Duration in seconds = (Total bytes / (Sample rate * Bit depth / 8 * Number of channels))` * `RTF` * `x%`
